src/parsers/manifest/dash/get_rep_availability_time_offset.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IRepresentationChildren } from "./node_parsers/Representation";
+
+/**
+ * Get addition of period availability time offset and representation
+ * availability time offset
+ * @param {number} adaptationAvailabilityTimeOffset
+ * @param {Object}
+ * @returns {number}
+ */
+function getRepAvailabilityTimeOffset(adaptationAvailabilityTimeOffset: number,
+                                      { baseURL,
+                                        segmentBase,
+                                        segmentTemplate }: IRepresentationChildren
+): number {
+  const fromRepBaseURL: number =
+    baseURL == null || baseURL.attributes.availabilityTimeOffset == null ?
+      0 : baseURL.attributes.availabilityTimeOffset;
+  const fromRepIndex: number = (() => {
+    if (segmentBase !== undefined) {
+      return segmentBase.availabilityTimeOffset !== undefined ?
+        segmentBase.availabilityTimeOffset : 0;
+    } else if (segmentTemplate !== undefined) {
+      return segmentTemplate.availabilityTimeOffset !== undefined ?
+        segmentTemplate.availabilityTimeOffset : 0;
+    }
+    return 0;
+  })();
```

Comment:
What if there is both ?

Also to check if it works but I personally find it more readable that way:
```ts
const fromRepIndex = segmentBase?.availabilityTimeOffset ??
                     segmentTemplate?.availabilityTimeOffset ??
                     0;
```

Reply:
Done